application/lib/catalog.dart

```diff
@@ -49,13 +50,15 @@ class _CatalogState extends State<Catalog> with SingleTickerProviderStateMixin {
         return FloatingActionButton(
           onPressed: () {
             LocationsDisposition.of(context).add(Location());
+            save('state.json', RootDisposition.last);
```

Comment:
we should only have one place that does this. the others can notify that place that there's something to do, but we don't want to duplicate 'state.json' and so on everywhere.

Easiest solution would be for the dispositions to have a reference to their parent and mark their parent dirty or some such, and if the root is dirty it schedules a save. or fires a callback and the main.dart does the save.